subsys/net/ip/route.c

```diff
@@ -729,6 +729,55 @@ int net_route_foreach(net_route_cb_t cb, void *user_data)
 static
 struct net_route_entry_mcast route_mcast_entries[CONFIG_NET_MAX_MCAST_ROUTES];
 
+static int mcast_route_iface_lookup(struct net_route_entry_mcast *entry, struct net_if *iface)
+{
+	for (int i = 0; i < CONFIG_NET_MCAST_ROUTE_MAX_IFACES; ++i) {
+		if (entry->ifaces[i] == iface) {
+			return i;
+		}
+	}
+
+	return -1;
+}
+
+bool net_route_mcast_iface_add(struct net_route_entry_mcast *entry, struct net_if *iface)
+{
+	if (!net_if_flag_is_set(iface, NET_IF_FORWARD_MULTICASTS)) {
+		return false;
+	}
+
+	if (mcast_route_iface_lookup(entry, iface) >= 0) {
+		/* Interface is already added */
+		return true;
+	}
+
+	for (int i = 0; i < CONFIG_NET_MCAST_ROUTE_MAX_IFACES; ++i) {
```

Comment:
Nit, there's a helper macro for these kind of arrays.

```c
	ARRAY_FOR_EACH(entry->ifaces, i) {
		//...
	}
```

Reply:
I have applied the remark and also added new first commit that uses the ARRAY_FOR_EACH_PTR() macro for all multicast routing related functions